tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem.cc

```diff
@@ -75,6 +78,25 @@ static void ParseGCSPath(absl::string_view fname, bool object_empty_ok,
   strcpy(*object, object_view.data());
 }
 
+class TempFile : public std::fstream {
+ public:
+  // We should specify openmode each time we call TempFile.
+  TempFile(const char* temp_file_name, std::ios::openmode mode)
```

Comment:
Since this won't get past the plugin boundary, we can use C++ types here everywhere. `string` might be better than `char*` due to memory management issues

Reply:
Done